### PR TITLE
Speed up Profiler::avg by a factor of 10

### PR DESCRIPTION
**Pull request by gregorycu**
Feedback in GitHub only please.

### CODE REVIEW

File: src/profiler.h [hunk 1]
```diff
@@ -69,23 +69,11 @@ class Profiler
...
-			if(n == m_data.end())
-				m_data[name] = value;
-			else
-				n->second += value;
-		}
+		int& count = m_avgcounts[name];
+		
+		assert(count != -2);
+		count = MYMAX(count, 0) + 1;
+		m_data[name] += value;
```
Comment (sfan5) [inline src/profiler.h]: Does this work when there is not entry for `name`?

File: src/profiler.h [hunk 2]
```diff
@@ -69,23 +69,11 @@ class Profiler
 	void avg(const std::string &name, float value)
 	{
 		JMutexAutoLock lock(m_mutex);
-		{
-			std::map<std::string, int>::iterator n = m_avgcounts.find(name);
-			if(n == m_avgcounts.end())
-				m_avgcounts[name] = 1;
```
Comment (sfan5) [inline src/profiler.h]: The new code does not do this.

Comment (gregorycu) [inline src/profiler.h]: What is "this"?

I assume you mean initialize to 1?

int& count = m_avgcounts[name]; // Will initialise to 0
assert(count != -2);
count = MYMAX(count, 0) + 1; // Will get incremented to 1

File: src/profiler.h (hunk 1 above)
Comment (gregorycu) [inline src/profiler.h]: m_avgcounts[name] will insert an element if it doesn't exist and initialise it to the default value, which is T(). This is 0 for number types.

**Comment by sfan5** [comment]
:+1:

**Comment by Zeno-** [comment]
Author will not discuss on IRC (marking as WIP; needs unit test)

**Comment by gregorycu** [comment]
I'll add a unit test, if that is required.

**Comment by gregorycu** [comment]
I have added a unit test as requested.

**Comment by gregorycu** [comment]
I have amended the unit test to factor in comments from ShadowNinja

**Comment by gregorycu** [comment]
Not likely to ever be merged. Closing.

**Comment by ShadowNinja** [comment]
Not a valid closing reason.

**Comment by est31** [comment]
only issue preventing merge is missing rebase? or something else too?

**Comment by ShadowNinja** [comment]
267c9f4cb4616afcf07a2a33aaca43a903ac895a

**Pull request closed without merging**